### DISCUSSION TITLE
Does WarpX support cylindrical geometry?

### DISCUSSION THREAD

**Discussion by griffinfarrow** [BLAST-WarpX/warpx#3321] [Q&A] [answered]
Hi there!
I'm trying to use WarpX for my work, and am ideally after a PIC code that works in 2D r-z geometry (for both particles and fields). Is this something WarpX includes?
I know that the documentation says that there is an azimuthal mode decomposition, but Smilei provides something similar and the fields are cylindrical even though the particles can only move in Cartesian geometry. In WarpX, do the particles move in cylindrical geometry too?

Many thanks!

**Comment by dpgrote** [comment, marked as answer]
WarpX uses the same type of model, where the particles move in 3D Cartesian space and the fields are only calculated on the radial grid (r-z space). When the particles interact with the grid, the particles's Cartesian coordinates are converted to Cylindrical coordinates.
